src/dotnet/commands/dotnet-projectmodel-server/Messengers/GlobalErrorMessenger.cs

```diff
@@ -16,9 +16,12 @@ protected override bool CheckDifference(ProjectSnapshot local, ProjectSnapshot r
             return remote != null && Equals(local.GlobalErrorMessage, remote.GlobalErrorMessage);
         }
 
-        protected override object CreatePayload(ProjectSnapshot local)
+        protected override void SendPayload(ProjectSnapshot local, Action<object> send)
         {
-            return local.GlobalErrorMessage;
+            if (local.GlobalErrorMessage != null)
```

Comment:
Should we just put this in ConnectionContext? Is there ever a reason you'd want to allow sending a null message?

This pattern of calling "SendPayload" and passing in a sender just to do a null check seems odd.